server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -444,17 +441,17 @@ public ActionModule(Settings settings, IndexNameExpressionResolver indexNameExpr
             UnaryOperator<RestHandler> newRestWrapper = plugin.getRestHandlerWrapper(threadPool.getThreadContext());
             if (newRestWrapper != null) {
                 logger.debug("Using REST wrapper from plugin " + plugin.getClass().getName());
+                if (plugin.getClass().getCanonicalName() == null ||
+                    plugin.getClass().getCanonicalName().startsWith("org.elasticsearch.xpack") == false) {
```

Comment:
Nit: I probably missed the discussion about why we prefer `getCanonicalName` over just `getName` here. But it seems to me that `getName` is more consistent since that is what we report in the error message (and also the debug log above)?